graders/python/python_autograder/pl_execute.py

```diff
@@ -77,7 +77,7 @@ def execute_code(
             str_student = pl_helpers.extract_ipynb_contents(f, ipynb_key)
         else:
             str_student = f.read()
-    str_student = str_leading + str_student + str_trailing
+    str_student = str_leading + "\n" + str_student + "\n" + str_trailing
```

Comment:
Would it be less surprising to users to only add the `\n` if the previous string was not already newline terminated? And should we also add a final newline on the very end (after `str_trailing`)?